piqe_ocp_lib/api/resources/ocp_cluster_versions.py

```diff
@@ -39,37 +44,163 @@ def update_cluster_version(self, cv_body):
 
-    def get_cluster_id(self):
+    def get_cluster_id(self) -> str:
         """
         Get cluster ID
-        :return: (str) cluster_id on success or None on failure
+        :return: cluster_id on success or None on failure
         """
-        cluster_id = None
         cluster_version_response = self.get_cluster_version()
-        for item in cluster_version_response.items:
-            if item["metadata"]["name"] == "version" and "clusterID" in dict(item["spec"]):
-                cluster_id = item["spec"]["clusterID"]
-        logger.info("Cluster ID : %s", cluster_id)
+        return cluster_version_response.spec.clusterID
+
+    def _build_spec(self, cv_body: Dict) -> Dict:
+        """
+        Add required fields in the user's input body.
+        :return: Updated dict with required values
+        """
+        cluster_version = self.get_cluster_version()
+
+        if not cluster_version:
+            return cv_body
+
+        cv_body.update({"apiVersion": self.api_version, "kind": self.kind})
+        metadata = {
+            "name": CLUSTER_VERSION_OPERATOR_ID,
+            "resourceVersion": cluster_version.metadata.resourceVersion,  # Required for successfull API call
+        }
+        spec = {"clusterId": cluster_version.spec.clusterID}  # Required for successfull API call
+
+        cv_body["metadata"] = {**cv_body.get("metadata", {}), **metadata}
+        cv_body["spec"] = {**cv_body.get("spec", {}), **spec}
+        return cv_body
+
+    def available_updates(self, channel: str = None) -> List[str]:
+        """
+        Get available versions to update
+        :return: list of available versions
+        """
+        cluster_version = self.get_cluster_version()
+        available_updates = cluster_version.status.availableUpdates
+
+        if not isinstance(available_updates, list):
+            # API returns null instead of empty list
+            logger.debug("No update available.")
+            return []
+
+        if channel:
+            available_updates = filter(lambda x: channel in x["channels"], available_updates)
+
+        return [update["version"] for update in available_updates]
+
+    def available_channels(self, kind: str = None) -> Set[str]:
+        """
+        Get available channels to upgrade
+        :return: set of available channels
+        """
+        cluster_version = self.get_cluster_version()
+        available_channels = cluster_version.status.availableUpdates
+
+        if not isinstance(available_channels, list):
+            # API returns null instead of empty list
+            logger.debug("No update available.")
+            return set()
+
+        if kind:
+            available_channels = filter(
+                lambda updates: any(kind in channel for channel in updates["channels"]),
+                available_channels
+            )
+
+        return set().union(*[e["channels"] for e in available_channels])
```

Comment:
Similar to the previous comment, to separate the logic for readability.

Reply:
I think this is okay for now, as it is working. We can make any agreed upon revisions in a future PR.